src/ngraph/ngraph_graph.cc

```diff
@@ -360,7 +360,23 @@ void CollapseSubgraph(Graph* graph, int subgraph_num) {
           tmpGraph->inputs_.emplace_back(input);
       }
     }
-
+    for (auto input : tmpGraph->inputs_) {
+      tmpGraph->input_is_weight_.push_back(false);
+    }
+    for (auto node : tmpGraph->nodes_) {
+      if (node->operation_ == "Convolution" ||
```

Comment:
not sure we want to qualify this logic with certain ops. i do not have better idea, other than to do it generally for all.

Reply:
I did these two ops because that's all that matters for the current optimizations. We need to make sure we're not memoizing layout conversions on data or op results. Data are always the first argument for these ops, so it's simple, but there are a lot of ops that can have data inputs in multiple places, thus the conservative first pass. If we can find a more general solution, that would be great.